Move WaitOnConnect() to the handler of wxSOCKET_CONNECTION to show that
wxSocketClient::m_connected is never set

Signed-off-by: Anders Larsen <al@alarsen.net>
---

 samples/sockets/client.cpp |   24 +++++++++++++-----------
 1 files changed, 13 insertions(+), 11 deletions(-)


diff --git a/samples/sockets/client.cpp b/samples/sockets/client.cpp
index bb30af2..4b3b3d8 100644
--- a/samples/sockets/client.cpp
+++ b/samples/sockets/client.cpp
@@ -385,16 +385,9 @@ void MyFrame::OpenConnection(wxSockAddress::Family family)
 
   m_text->AppendText(_("\nTrying to connect (timeout = 10 sec) ...\n"));
   m_sock->Connect(*addr, false);
-  m_sock->WaitOnConnect(10);
 
   if (m_sock->IsConnected())
     m_text->AppendText(_("Succeeded ! Connection established\n"));
-  else
-  {
-    m_sock->Close();
-    m_text->AppendText(_("Failed ! Unable to connect\n"));
-    wxMessageBox(_("Can't connect to the specified host"), _("Alert !"));
-  }
 
   UpdateStatusBar();
 }
@@ -687,10 +680,19 @@ void MyFrame::OnSocketEvent(wxSocketEvent& event)
 
   switch(event.GetSocketEvent())
   {
-    case wxSOCKET_INPUT      : s.Append(_("wxSOCKET_INPUT\n")); break;
-    case wxSOCKET_LOST       : s.Append(_("wxSOCKET_LOST\n")); break;
-    case wxSOCKET_CONNECTION : s.Append(_("wxSOCKET_CONNECTION\n")); break;
-    default                  : s.Append(_("Unexpected event !\n")); break;
+    case wxSOCKET_INPUT:
+      s.Append(_("wxSOCKET_INPUT\n"));
+      break;
+    case wxSOCKET_LOST:
+      s.Append(_("wxSOCKET_LOST\n"));
+      break;
+    case wxSOCKET_CONNECTION:
+      s.Append(_("wxSOCKET_CONNECTION\n"));
+      m_sock->WaitOnConnect(0);
+      break;
+    default:
+      s.Append(_("Unexpected event !\n"));
+      break;
   }
 
   m_text->AppendText(s);
